### ISSUE TITLE
Browser doesn't load testing page

### ISSUE THREAD

**Issue by MAKMED1337** [FlareSolverr/FlareSolverr#1403]
### Have you checked our README?

- [X] I have checked the README

### Have you followed our Troubleshooting?

- [X] I have followed your Troubleshooting

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you checked the discussions?

- [X] I have read the Discussions

### Have you ACTUALLY checked all these?

YES

### Environment

```markdown
- FlareSolverr version: 3.3.21
- Last working FlareSolverr version: -
- Operating system: Linux archlinux 6.11.5-zen1-1-zen
- Are you using Docker: no
- FlareSolverr User-Agent (see log traces or / endpoint): doesn't get to this point
- Are you using a VPN: no
- Are you using a Proxy: no
- Are you using Captcha Solver: no
- If using captcha solver, which one: -
- URL to test this issue: https://google.com
```

### Description

This issue is similar to https://github.com/FlareSolverr/FlareSolverr/issues/1384, but it is another bug.
When using docker in a headless mode it works, but using it on the desktop **without** headless through `python src/flaresolverr.py` starts the browser and then simply doesn't load anything.

Can be helpful: ~~I'm using linux, hyprland~~ it doesn't work on both linux and windows, x64 processor, I have tested it on an nvidia gpu, and intel's integrated gpu.

I found that removing `options.add_argument('--disable-software-rasterizer')` in `src/utils.py` works for me, but I don't know is it only a mine problem, or a problem that occurs on x86-64 CPUs.

### Logged Error Messages

```text
2024-10-28 11:19:14 INFO     FlareSolverr 3.3.21
2024-10-28 11:19:14 INFO     Testing web browser installation...
2024-10-28 11:19:14 INFO     Platform: Linux-6.11.5-zen1-1-zen-x86_64-with-glibc2.40
2024-10-28 11:19:14 INFO     Chrome / Chromium path: /usr/bin/chromium
2024-10-28 11:19:14 INFO     Chrome / Chromium major version: 130
2024-10-28 11:19:14 INFO     Launching web browser...

And then an infinite load.
```

### Screenshots

![image](https://github.com/user-attachments/assets/881ef568-c73b-4c04-b21a-af6e97648927)

**Comment by vcmania** [comment]
how about this style?
check browser_executable_path

utils.py
    # detect chrome path
    browser_executable_path = get_chrome_exe_path()
    logging.debug('browser_executable_path: %s', browser_executable_path)
    if os.name == 'nt':
        if browser_executable_path.startswith('C:\\Program Files (x86)\\Google\\Chrome\\Application\\chrome.exe') == False:
            options.add_argument('--disable-software-rasterizer')
    
    # downloads and patches the chromedriver
    # if we don't set driver_executable_path it downloads, patches, and deletes the driver each time
    try:
        driver = uc.Chrome(options=options, browser_executable_path=browser_executable_path,
                           driver_executable_path=driver_exe_path, version_main=version_main,
                           windows_headless=windows_headless, headless=get_config_headless())
    except Exception as e:
        logging.error("Error starting Chrome: %s" % e)

**Comment by MAKMED1337** [comment]
I've checked and this bug also occurs on a windows machine

**Comment by MAKMED1337** [comment]
Considering the comment above the line, I guess the issue is only on a non-arm CPU, to solve this I guess we can add some if about `platform.machine()`, but I'm not sure about the correct string to put there: https://stackoverflow.com/questions/45125516/possible-values-for-uname-m

**Comment by vcmania** [comment]
If the problem only occurs on non-arm CPUs, wouldn't it be okay to do something like this?

utils.py (around line 140)

    # attempt to fix Docker ARM32 build
    options.add_argument('--disable-gpu-sandbox')
    # ----------------------------------------------------------------
    # for only arm architecture --disable-software-rasterizer
    # options.add_argument('--disable-software-rasterizer')
    arm_architectures = ['arm', 'aarch']
    architecture = platform.system()
    for arch in arm_architectures:
        if arch.startswith(architecture):
            options.add_argument('--disable-software-rasterizer')
    # ----------------------------------------------------------------
    options.add_argument('--ignore-certificate-errors')

**Comment by MAKMED1337** [comment]
Sounds like a good solution, you can make a PR with this.

**Comment by vcmania** [comment]
Sorry.
I uploaded it without testing it.
You need to change "platform.system()" to "platform.machine()".

**Comment by vcmania** [comment]
I think you can change it simply like below.

    IS_ARMARCH = platform.machine().startswith(("arm", "aarch"))
    if IS_ARMARCH:
        options.add_argument('--disable-software-rasterizer')

**Comment by MAKMED1337** [comment]
Will you make a pull request ?

**Comment by vcmania** [comment]
I am not a contributor so I cannot make pull requests.
Please fork your copy and make your changes.

**Comment by TheRedCyclops** [comment]
it is also happening for me on amd64 arch

**Comment by MAKMED1337** [comment]
> it is also happening for me on amd64 arch

Does my fix(PR) help ?

**Comment by konjas** [comment]
I had the same issue. The PR proposed by @MAKMED1337 solved it.

**Comment by svaningelgem** [comment]
I have the same issue, but with these settings:
- Windows 10
- I took the chromium driver of playwright (chrome 131.0.6778.33) and dropped it in the src-directory so it could be found
- started with `python src\flaresolverr.py`
Just like the OP: infinite load

Then I removed this chrome directory from the src dir, and flaresolverr uses my installed one in program files (also 131), same thing.

Then I downloaded the win-release and took the 123 chrome driver that was included there and dropped it in the src directory
--> No more infinite load.

The (proposed fix above)[https://github.com/FlareSolverr/FlareSolverr/issues/1403#issuecomment-2448896535] didn't solve my issue.

The problem is also not with starting the browser, as I see it running in the task manager.

So something changed (at least on windows) between chrome 123 & 131 that makes it not able to interface.
